ping/rust/Dockerfile

```diff
@@ -4,43 +4,23 @@ WORKDIR /usr/src/testplan
 RUN apt-get update && apt-get install -y cmake protobuf-compiler
 
 ARG PLAN_PATH="./"
-
-# Cache dependencies between test runs, see testground examples.
-RUN mkdir -p ./plan/src/
-RUN echo "fn main() { println!(\"If you see this message, you may want to clean up the target directory or the Docker build cache.\") }" > ./plan/src/main.rs
-COPY ./plan/${PLAN_PATH}/Cargo.lock ./plan/${PLAN_PATH}/Cargo.toml ./plan/
-
-RUN cd ./plan/ && \
-        cargo update --dry-run && \
-        cargo fetch && \
-        # use a default feature to rely on docker caching.
-        cargo build --features "libp2pv0450"
-
-ARG CARGO_PATCH=""
-ARG CARGO_REMOVE=""
-
-RUN if [ ! -z "${CARGO_REMOVE}" ]; then sed -i "s/^${CARGO_REMOVE}.*//" ./plan/Cargo.toml; fi
-RUN echo "${CARGO_PATCH}" >> ./plan/Cargo.toml
-
-# Backup Cargo file to preserve patches.
-RUN cp ./plan/Cargo.toml ./plan/Cargo.lock /tmp/
-
 COPY ./plan/${PLAN_PATH} ./plan
+RUN cd ./plan/ && cargo build  # Initial build acts as a cache.
 
-# This is in order to make sure `main.rs`s mtime timestamp is updated to avoid the dummy `main`
-# remaining in the binary.
-# https://github.com/rust-lang/cargo/issues/9598
-RUN touch ./plan/src/main.rs
-
-RUN mv /tmp/Cargo.toml /tmp/Cargo.lock ./plan
+ARG GIT_TARGET=""
+RUN if [ ! -z "${GIT_TARGET}" ]; then sed -i "s,^git.*,git = \"https://${GIT_TARGET}\"," ./plan/Cargo.toml; fi
 
-ARG CARGO_FEATURES=""
+ARG GIT_REF=""
+RUN if [ ! -z "${GIT_REF}" ]; then sed -i "s/^rev.*/rev = \"${GIT_REF}\"/" ./plan/Cargo.toml; fi
 
-RUN cd ./plan/ && \
-        cargo build --features="${CARGO_FEATURES}"
+ARG BINARY_NAME
+RUN cd ./plan/ && cargo build --bin=${BINARY_NAME} # Cargo will update lockfile on changed manifest (i.e. if one of the above `sed`s patched it)
+RUN ls -la /usr/src/testplan/plan/target/debug/
+RUN echo ${BINARY_NAME}
 
 FROM debian:bullseye-slim
-COPY --from=builder /usr/src/testplan/plan/target/debug/testplan /usr/local/bin/testplan
+COPY --from=builder /usr/src/testplan/plan/target/debug/${BINARY_NAME} /usr/local/bin/
+RUN mv /usr/local/bin/${BINARY_NAME} /usr/local/bin/testplan
```

Comment:
Is `BINARY_NAME` defined here?

CI error:
```
Step 17/20 : RUN mv /usr/local/bin/${BINARY_NAME} /usr/local/bin/testplan
 ---> Running in 66b2252d5e55
mv: cannot move '/usr/local/bin/' to a subdirectory of itself, '/usr/local/bin/testplan'
```

Reply:
Oh my god! The ARG definition from above doesn't carry over! That is the bug! Thanks haha!